### PR TITLE
chore: Refactor DefCollector duplicate errors

### PR DESCRIPTION
**Pull request by yordanmadzhunkov**
# Description
Def Collector issues several errors that has to do with duplicates. I united them into one common error type: Duplicate. It will be used in next steps when def collector collects trait definitions. This changes can be merged without conflicts, so I separated them into their own PR. 

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by yordanmadzhunkov** [comment]
@jfecher

File: crates/noirc_frontend/src/hir/def_collector/dc_mod.rs [hunk 1]
```diff
@@ -172,7 +180,11 @@ impl<'a> ModCollector<'a> {
                 self.def_collector.def_map.modules[self.module_id.0].declare_struct(name, id);
 
             if let Err((first_def, second_def)) = result {
-                let err = DefCollectorErrorKind::DuplicateFunction { first_def, second_def };
+                let err = DefCollectorErrorKind::Duplicate {
+                    typ: "type definition".to_string(),
```
Comment (yordanmadzhunkov) [inline crates/noirc_frontend/src/hir/def_collector/dc_mod.rs:184]: Notice that in the old implementation error was DuplicateFunction, but it should be duplicate type definition !

File: crates/noirc_frontend/src/hir/def_collector/errors.rs [hunk 2]
```diff
@@ -8,14 +8,8 @@ use thiserror::Error;
...
-    #[error("duplicate function found in namespace")]
-    DuplicateFunction { first_def: Ident, second_def: Ident },
-    #[error("duplicate function found in namespace")]
-    DuplicateModuleDecl { first_def: Ident, second_def: Ident },
-    #[error("duplicate import")]
-    DuplicateImport { first_def: Ident, second_def: Ident },
-    #[error("duplicate global found in namespace")]
-    DuplicateGlobal { first_def: Ident, second_def: Ident },
+    #[error("duplicate {typ} found in namespace")]
+    Duplicate { typ: String, first_def: Ident, second_def: Ident },
```
Comment (jfecher) [inline crates/noirc_frontend/src/hir/def_collector/errors.rs:12]: Nitpick: can we change `typ` to a `&'static str`?

Comment (yordanmadzhunkov) [reply crates/noirc_frontend/src/hir/def_collector/errors.rs:12]: yes, I was wondering which is better, I will go with your suggestion and update the PR tomorrow :)

Comment (yordanmadzhunkov) [reply crates/noirc_frontend/src/hir/def_collector/errors.rs]: Done

**Review by jfecher — APPROVED** [review]
`DuplicateType` should impl `Display` instead of `Debug` but I'll fix that in a subsequent PR

**Pull request merged**